src/pages/nft/class/_classId/_nftId.vue

```diff
@@ -104,17 +77,86 @@
                 v-for="m in messageList"
                 :key="m.txHash"
                 :type="m.event"
-                :tx-hash="m.txHash"
                 :from-type="m.fromType"
                 :from-wallet="m.fromWallet"
                 :to-type="m.toType"
                 :to-wallet="m.toWallet"
                 :message="m.message"
                 :message-type="m.messageType"
+                :is-list="true"
                 tag="li"
               />
             </ul>
           </div>
+
+          <!-- NFT Preview -->
+          <div class="flex flex-col items-center col-span-1">
+            <NFTGemWrapper :collected-count="collectedCount">
+              <NFTPagePreviewCard
+                :url="NFTExternalUrl"
+                :image-bg-color="NFTImageBackgroundColor"
+                :image-url="NFTImageUrl"
+                :avatar-url="creatorAvatar"
+                :avatar-size="40"
+                :is-avatar-outlined="isCreatorCivicLiker"
+                :iscn-owner="iscnOwner"
+                :display-name="creatorDisplayName"
+                :nft-name="NFTName"
+                :nft-description="NFTDescription"
+                :nft-price="NFTPrice"
+                @collect="handleCollectFromPreviewSection"
+                @view-content="handleViewContent"
+              />
+            </NFTGemWrapper>
+            <ButtonV2
+              class="mt-[24px] text-medium-gray"
+              content-class="text-[12px]"
+              preset="plain"
+              size="mini"
+              :to="{ name: 'nft-class-classId', params: { classId } }"
+            >
+              <template #prepend>
+                <IconEye class="w-[12px] h-[12px]" />
+              </template>
+              {{ $t('nft_details_page_button_view_details') }}
+            </ButtonV2>
+          </div>
+
+          <!-- NFT Collect CTA -->
+          <div
+            v-if="nftIsNew"
+            class="flex flex-col items-center col-span-2 py-[40px] gap-[24px]"
+          >
+            <h2 class="text-[32px] leading-1_5 text-center">{{ $t('nft_share_page_creator_title') }}</h2>
+            <div
+              v-if="nftCreatorMessage"
+              class="py-[40px] px-[24px] w-full"
+            >
+              <NFTMessage
+                class="w-full"
+                type="purchase"
+                from-type="creator"
+                :from-wallet="iscnOwner"
+                :message="nftCreatorMessage"
```

Comment:
in API nftCreatorMessage is a template string that has a `.replace` element (currently not used), should we implement it in client also?

Reply:
Implement? Isn't the return memo already parsed?

Reply:
but `nftCreatorMessage` is just `return this.NFTClassMetadata?.message || '';` here without parsing?

Reply:
I thought the parsing is done during mint, am I getting it wrong?

Reply:
Wait, I got it, the parsing is done during collect not mint, so this shows the preview of the message, right?

Reply:
yes, parsing is done during transaction and result is written into memo
so if we really start using template string in message, we might need to implement parsing here too
or we can also ignore this too, the template string is kind of readable

Reply:
We should put the logic into one lib when we develop this further